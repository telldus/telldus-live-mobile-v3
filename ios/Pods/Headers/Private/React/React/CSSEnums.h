../../../../../../node_modules/react-native/React/CSSLayout/CSSEnums.h