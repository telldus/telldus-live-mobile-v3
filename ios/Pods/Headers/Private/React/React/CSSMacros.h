../../../../../../node_modules/react-native/React/CSSLayout/CSSMacros.h