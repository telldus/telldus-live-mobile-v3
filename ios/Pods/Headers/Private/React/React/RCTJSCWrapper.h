../../../../../../node_modules/react-native/React/Executors/RCTJSCWrapper.h