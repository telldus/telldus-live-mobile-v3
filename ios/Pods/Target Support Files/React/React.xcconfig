CLANG_CXX_LANGUAGE_STANDARD = c++14
CONFIGURATION_BUILD_DIR = $PODS_CONFIGURATION_BUILD_DIR/React
GCC_PREPROCESSOR_DEFINITIONS = $(inherited) COCOAPODS=1
HEADER_SEARCH_PATHS = "${PODS_ROOT}/Headers/Private" "${PODS_ROOT}/Headers/Private/React" "${PODS_ROOT}/Headers/Public" "${PODS_ROOT}/Headers/Public/React"
OTHER_LDFLAGS = -l"stdc++" -framework "JavaScriptCore"
PODS_BUILD_DIR = $BUILD_DIR
PODS_CONFIGURATION_BUILD_DIR = $PODS_BUILD_DIR/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)
PODS_ROOT = ${SRCROOT}
PRODUCT_BUNDLE_IDENTIFIER = org.cocoapods.${PRODUCT_NAME:rfc1034identifier}
SKIP_INSTALL = YES
