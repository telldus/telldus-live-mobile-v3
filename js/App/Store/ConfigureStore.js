/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 */

'use strict';

import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import promise from './Promise';
import array from './Array';
import analytics from './Analytics';
import reducers from 'Reducers';
import createLogger from 'redux-logger';
import { persistStore, autoRehydrate } from 'redux-persist';
import { AsyncStorage } from 'react-native';

let isDebuggingInChrome = __DEV__ && !!window.navigator.userAgent;

let logger = createLogger({
	predicate: (getState, action) => isDebuggingInChrome,
	collapsed: true,
	duration: true,
});

let createTheStore = applyMiddleware(thunk, promise, array, analytics, logger)(createStore);

let _store;
export function configureStore(onComplete: ?() => void) {
	const store = autoRehydrate()(createTheStore)(reducers);
	persistStore(store, {storage: AsyncStorage}, onComplete);
	if (isDebuggingInChrome) {
		window.store = store;
	}
	_store = store; // TODO: fix this ugly stuff
	return store;
}
export function getStore() { return _store; }
