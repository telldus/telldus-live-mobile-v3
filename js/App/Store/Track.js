/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 */

'use strict';

import type { Action } from '../actions/types';
import DeviceInfo from 'react-native-device-info';

import { Analytics, Hits as GAHits } from 'react-native-google-analytics';
import { googleAnalyticsId } from 'Config';

function track(action: Action): void {

	let clientId = DeviceInfo.getUniqueID();
	let ga = new Analytics(googleAnalyticsId, clientId, 1, DeviceInfo.getUserAgent());

	if (action.type === 'SWITCH_TAB') {
		let screenView = new GAHits.ScreenView(
			'Telldus Live! app',
			action.tab,
			DeviceInfo.getVersion(),
			DeviceInfo.getBundleId()
		);
		ga.send(screenView);
	} else {
		let gaEvent = new GAHits.Event('Action', action.type);
		ga.send(gaEvent);
	}
}

module.exports = track;
