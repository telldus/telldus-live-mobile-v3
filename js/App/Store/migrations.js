/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

// @flow

'use strict';

import {
	DEFAULT_DASHBOARD_ID,
} from '../Lib/dashboardUtils';

export default function migrations(state: Object = {}): Promise<any> {
	const { tabs, ...withOutTabs } = state;

	let newState = {...state};
	// Remove redux store object 'tabs' if present.
	if (tabs) {
		newState = withOutTabs;
	}

	const { App, ...withOutApp } = newState;
	// Rename redux store object 'App' to 'app'.
	if (App) {
		newState = {
			...withOutApp,
			app: App,
		};
	}

	const { app, dashboard, user } = newState;
	if (app && !app.defaultSettings) {
		newState = {
			...newState,
			app: {
				...app,
				defaultSettings: {
					dimmerSensitivity: 5,
					sortingDB: 'Chronological',
					activeDashboardId: DEFAULT_DASHBOARD_ID,
				},
			},
		};
	}

	if (app && app.defaultSettings && !app.defaultSettings.activeDashboardId) {
		newState = {
			...newState,
			app: {
				...app,
				defaultSettings: {
					...app.defaultSettings,
					activeDashboardId: DEFAULT_DASHBOARD_ID,
				},
			},
		};
	}

	if (user && !user.activeDashboardId) {
		newState = {
			...newState,
			user: {
				...user,
				activeDashboardId: DEFAULT_DASHBOARD_ID,
			},
		};
	}

	let { userId, userProfile = {} } = user || {};
	if (!userId) {
		const {
			uuid,
		} = userProfile;
		userId = uuid;
	}

	if (dashboard && userId) {
		const {
			devicesById = {},
			sensorsById = {},
			sensorIds,
			deviceIds,
		} = dashboard;

		const prevDataType = [true, false];

		let newDashboard = {...dashboard};

		const devicesByIdC = Object.keys(devicesById);
		if (devicesByIdC.length > 0 && prevDataType.indexOf(devicesById[devicesByIdC[0]]) !== -1) {
			newDashboard = {
				...newDashboard,
				devicesById: {
					[userId]: {
						[DEFAULT_DASHBOARD_ID]: devicesById,
					},
				},
				deviceIds: {
					[userId]: {
						[DEFAULT_DASHBOARD_ID]: deviceIds,
					},
				},
			};
		}
		const sensorsByIdC = Object.keys(sensorsById);
		if (sensorsByIdC.length > 0 && prevDataType.indexOf(sensorsById[sensorsByIdC[0]]) !== -1) {
			newDashboard = {
				...newDashboard,
				sensorsById: {
					[userId]: {
						[DEFAULT_DASHBOARD_ID]: sensorsById,
					},
				},
				sensorIds: {
					[userId]: {
						[DEFAULT_DASHBOARD_ID]: sensorIds,
					},
				},
			};
		}
		newState = {
			...newState,
			dashboard: {
				...newDashboard,
			},
		};
	}

	const { gateways, sensorsList } = newState;
	if (gateways) {
		// Insert the attribute/property 'localKey' to each gateways object if not already present.
		const newGateways = insertLocalKeyAttribute(gateways);
		newState = {
			...newState,
			gateways: newGateways,
		};
	}

	if (sensorsList && sensorsList.defaultTypeById) {
		const defaultSensorSettings = migrateToDefaultSensorSettings(sensorsList.defaultTypeById);
		newState = {
			...newState,
			sensorsList: {
				defaultSensorSettings,
			},
		};
	}

	if (sensorsList && sensorsList.defaultSensorSettings) {
		let newDefaultSensorSettings = {};
		let { defaultSensorSettings } = sensorsList;
		for (let key in defaultSensorSettings) {
			const { historySettings = {}, ...others } = defaultSensorSettings[key];
			const { selectedTwo, selectedOne, ...othersH } = historySettings;
			if ((selectedTwo && selectedTwo.type === 'wdir') || (selectedOne && selectedOne.type === 'wdir')) {
				newDefaultSensorSettings[key] = {
					...others,
					historySettings: {
						...othersH,
						selectedTwo: undefined,
						selectedOne: undefined,
					},
				};
			} else {
				newDefaultSensorSettings[key] = defaultSensorSettings[key];
			}
		}

		newState = {
			...newState,
			sensorsList: {
				defaultSensorSettings: newDefaultSensorSettings,
			},
		};
	}

	return Promise.resolve(newState);
}


const insertLocalKeyAttribute = (gateways: Object): Object => {
	const { byId } = gateways;
	let newById = {};
	for (let key in byId) {
		let gateway = byId[key];
		if (gateway && !gateway.localKey) {
			gateway = {
				...gateway,
				localKey: {
					key: null,
					ttl: null,
					uuid: null,
					address: null,
					port: null,
					macAddress: null,
					supportLocal: false,
				},
			};
		}
		newById[key] = gateway;
	}
	return {
		...gateways,
		byId: newById,
	};
};


const migrateToDefaultSensorSettings = (defaultTypeById: Object): Object => {
	let defaultSensorSettings = {};
	for (let key in defaultTypeById) {
		defaultSensorSettings[key] = {
			displayType: defaultTypeById[key],
			displayTypeDB: null,
			historySettings: {},
		};
	}
	return defaultSensorSettings;
};
