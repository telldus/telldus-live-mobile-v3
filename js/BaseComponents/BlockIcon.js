/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 */

// @flow

'use strict';

import React, { Component } from 'react';
import View from './View';
import IconTelldus from './IconTelldus';
import Theme from '../App/Theme';

type DefaultProps = {
	color: string,
	bgColor: string,
	backgroundMask: boolean,
};

type Props = {
	icon?: string,
	size?: number,
	color?: string,
	bgColor?: string,
	style?: Array<any> | Object,
	containerStyle?: Array<any> | Object,
	backgroundMask?: boolean,
	backgroundMaskStyle?: Array<any> | Object,
	iconLevel?: number,
	blockLevel?: number,
};

export default class BlockIcon extends Component<Props, null> {
	props: Props;

	static defaultProps: DefaultProps = {
		color: '#fff',
		bgColor: Theme.Core.brandPrimary,
		backgroundMask: false,
	};

	render(): Object {
		const {
			style,
			containerStyle,
			icon,
			size,
			color,
			backgroundMask,
			backgroundMaskStyle,
			iconLevel,
			blockLevel,
		} = this.props;
		const defaultStyle = this._getDefaultStyle();

		return (
			<View
				level={blockLevel || 8}
				style={[defaultStyle, containerStyle]}>
				{backgroundMask && (<View style={backgroundMaskStyle}/>)}
				<IconTelldus
					level={iconLevel}
					icon={icon}
					size={size}
					color={typeof iconLevel === 'undefined' ? color : undefined}
					style={style}/>
			</View>
		);
	}

	_getDefaultStyle = (): Object => {
		return {
			alignItems: 'center',
			justifyContent: 'center',
			backgroundColor: this.props.bgColor,
		};
	};
}
