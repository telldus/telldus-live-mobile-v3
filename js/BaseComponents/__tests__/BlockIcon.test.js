/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope this it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import React from 'react';
import {
	Dimensions,
} from 'react-native';
import {act} from 'react-test-renderer';

import {
	rendererWithIntlAndReduxProviders,
} from '../../Utils/jestUtils';
import BlockIcon from '../BlockIcon';
import {
	setAppLayout,
} from '../../App/Actions';
import { configureStore } from '../../App/Store/ConfigureStore';

let {height, width} = Dimensions.get('window');

const store = configureStore().store;
const setLayout = () => {
	store.dispatch(setAppLayout({
		height,
		width,
	}));
};

describe('<BlockIcon /> - snapshot', () => {

	it('renders BlockIcon', () => {
		let component;
		act(() => {
			setLayout();
			component = rendererWithIntlAndReduxProviders(
				<BlockIcon icon={'info'}/>
			);
		});

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

});
