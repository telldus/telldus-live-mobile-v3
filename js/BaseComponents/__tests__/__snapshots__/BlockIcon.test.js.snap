// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockIcon /> - snapshot renders BlockIcon 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1b365d",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    importantForAccessibility="yes"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#fff",
        "fontFamily": "telldusicons",
        "fontSize": 12,
      }
    }
  >
    info
  </Text>
</View>
`;
